mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -87,24 +108,7 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	dep DepositT,
 ) error {
-	var nextDepositIndex uint64
-	switch depositIndex, err := st.GetEth1DepositIndex(); {
-	case err == nil:
-		// just increment the deposit index if no error
-		nextDepositIndex = depositIndex + 1
-	case sp.processingGenesis && err != nil:
-		// If errored and still processing genesis,
-		// Eth1DepositIndex may have not been set yet.
-		nextDepositIndex = 0
-	default:
-		// Failed retrieving Eth1DepositIndex outside genesis is an error
-		return fmt.Errorf(
-			"failed retrieving eth1 deposit index outside of processing genesis: %w",
-			err,
-		)
-	}
-
-	if err := st.SetEth1DepositIndex(nextDepositIndex); err != nil {
```

Comment:
btw, this was wrong conceptually. `Eth1DepositIndex` should hold the index of the latest deposit stored, not the expected next one, in order to adhere to Eth 2.0 specs.
Fixed in this PR (which was broken at some point due to `Eth1DepositIndex` being wrongly set)

Reply:
Do we need to keep thsi for bartio as a special case?

Reply:
Done for all good measures. I would cause issues if syncing from scratch. Boonet should be fine